### PR TITLE
Improve error messages for import-related error codes

### LINKED ISSUE #164: Error message for builtin generics is unclear for people unfamiliar with PEP 585
See https://github.com/python/typeshed/pull/7086

### PR DESCRIPTION
**Pull request by AlexWaygood**
Fixes #164.

@Spindel, what do you think of the proposed new error messages? :) You can see what the new errors would be by looking at the diff in `tests/imports.pyi`.

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]
Great!

**Comment by Spindel** [comment]
I think it's quite a lot better, and the fix with changing   `good_cls_name`  to `good_syntax` makes the other code better to read on it's own.

Thanks for this!

:strawberry:

**Comment by AlexWaygood** [comment]
> Thanks for this!
> 
> 🍓

No problem at all -- thanks for the review!

File: pyi.py [hunk 1]
```diff
@@ -53,58 +53,71 @@ class TypeVarInfo(NamedTuple):
...
+# We use `None` to signify that the object shouldn't  be parameterised.
+_BAD_Y023_IMPORTS = {
+    # collections.abc aliases
+    "Awaitable": "T",
+    "Coroutine": "YieldType, SendType, ReturnType",
+    "AsyncIterable": "T",
+    "AsyncIterator": "T",
+    "AsyncGenerator": "YieldType, SendType",
+    # typing aliases
+    "Protocol": "T",
```
Comment (Akuli) [inline pyi.py:108]:
```suggestion
    "Protocol": None,
```

File: pyi.py [hunk 2]
```diff
@@ -53,58 +53,71 @@ class TypeVarInfo(NamedTuple):
...
-    }
-)
+# We use `None` to signify that the object shouldn't  be parameterised.
+_BAD_Y023_IMPORTS = {
+    # collections.abc aliases
+    "Awaitable": "T",
+    "Coroutine": "YieldType, SendType, ReturnType",
+    "AsyncIterable": "T",
+    "AsyncIterator": "T",
+    "AsyncGenerator": "YieldType, SendType",
```
Comment (Akuli) [inline pyi.py:106]: Does this really not have a return type parameter like `Coroutine` and `Generator` have?

Comment (AlexWaygood) [reply pyi.py:106]: Yeah, it doesn't. From the [docs](https://docs.python.org/3/library/typing.html#typing.AsyncGenerator):

> Unlike normal generators, async generators cannot return a value, so there is no ReturnType type parameter.

**Pull request merged**